packages/core/src/system.ts

```diff
@@ -0,0 +1,34 @@
+import { ActorSystem, ActorSystemInfo, AnyActorRef } from './types.js';
+
+export function createSystem<T extends ActorSystemInfo>(): ActorSystem<T> {
+  let sessionIdIndex = 0;
+  const children = new Map<string, AnyActorRef>();
+  const keyedActors = new Map<keyof T['actors'], AnyActorRef | undefined>();
+  const reverseKeyedActors = new WeakMap<AnyActorRef, keyof T['actors']>();
+
+  const system: ActorSystem<T> = {
+    _register: (actorRef) => {
+      const id = `x:${sessionIdIndex++}`;
+      children.set(id, actorRef);
+      return id;
+    },
+    _unregister: (actorRef) => {
+      children.delete(actorRef.id);
+      const key = reverseKeyedActors.get(actorRef);
+
+      if (key !== undefined) {
+        keyedActors.delete(key);
+        reverseKeyedActors.delete(actorRef);
+      }
+    },
+    get: (key) => {
+      return keyedActors.get(key) as T['actors'][any];
+    },
+    set: (key, actorRef) => {
+      keyedActors.set(key, actorRef);
+      reverseKeyedActors.set(actorRef, key);
+    }
```

Comment:
What do you think about throwing if the `key` is already set?

Reply:
We should allow overriding the actor

Reply:
> Why though? It seems like an easy mistake to make. What is practical use case for a hard overwrite like this?

Because, for example, if a _spawned_ actor is assigned a `systemId` and we want to instead spawn a different actor for that `systemId` (replacing it), then it is not easy to replace it.

We can disallow it for now though.

Reply:
But what would be a practical use case for such replacing?

Reply:
> But what would be a practical use case for such replacing?

Not sure; I can disable this, we can always add it back later if we find a valid use-case.

Reply:
ba996b642612b51df2e7e45723da0cfb57e9f81d